tools/continuous_batching/limit_checker.py

```diff
@@ -0,0 +1,247 @@
+        for json_str in json_list:
+            result = json.loads(json_str)
+            retval["prompts"].append(result["prompt"])
+    return retval
+
+def get_scheduler_config(num_kv_blocks: Optional[int]) -> SchedulerConfig:
+    scheduler_config = SchedulerConfig()
+    if num_kv_blocks is not None:
+        scheduler_config.num_kv_blocks = num_kv_blocks
+    scheduler_config.dynamic_split_fuse = True
+    scheduler_config.max_num_batched_tokens = 32 * num_kv_blocks
+    scheduler_config.max_num_seqs = 256
+    scheduler_config.use_cache_eviction = False
+    return scheduler_config
+
+@dataclass
+class ConvertedModel:
+    model: OVModelForCausalLM
+    tokenizer: AutoTokenizer
+    models_path: Path
+
+
+def get_converted_model(base_model_path: Path, model_id: str):
+    model = retry_request(lambda: OVModelForCausalLM.from_pretrained(model_id, trust_remote_code=True, load_in_8bit=False, compile=False, ov_config=get_default_llm_properties()))
+    tokenizer = retry_request(lambda: AutoTokenizer.from_pretrained(model_id))
+    models_path = base_model_path / model_id
+    models_path.mkdir(parents=True, exist_ok=True)
+    model.save_pretrained(models_path)
+    ov_tokenizer, ov_detokenizer = convert_tokenizer(tokenizer, with_detokenizer=True, skip_special_tokens=True)
+    serialize(ov_tokenizer, models_path / "openvino_tokenizer.xml")
+    serialize(ov_detokenizer, models_path / "openvino_detokenizer.xml")
+    converted_model = ConvertedModel(model, tokenizer, models_path)
+    return converted_model
+
+
+SHORT_CACHE_EVICTION_CONFIG = CacheEvictionConfig(start_size=32, recent_size=32, max_cache_size=96, aggregation_mode=AggregationMode.NORM_SUM)
+LONGBENCH_CACHE_EVICTION_CONFIG = CacheEvictionConfig(start_size=32, recent_size=128, max_cache_size=672, aggregation_mode=AggregationMode.NORM_SUM)
+
+import openvino.properties.hint as hints
+import openvino.properties as props
+import openvino as ov
+
+def get_default_llm_properties():
+    return {
+        hints.inference_precision : ov.Type.f32,
+        hints.kv_cache_precision : ov.Type.f16,
+    }
+
+def run_and_write_metrics(model, prompt, generation_config, report_file):
+    result: GenerationResult = model_cb_opt.generate([prompt], generation_config=[generation_config])
+
+    pipeline_opt_metrics = model_cb_opt.get_metrics()
+    rss_usage_gb = psutil.Process(os.getpid()).memory_info().rss / 1024 ** 3
+    result_length = len(result[0].m_generation_ids[0])
+    print(f"avg_cache_usage:{pipeline_opt_metrics.avg_cache_usage:.2f}% max_cache_usage:{pipeline_opt_metrics.max_cache_usage:.2f}% rss_usage:{rss_usage_gb:.3f} GB")
+    print(f"result length: {result_length}")
+    print()
+
+    if args.report is not None:
+        with open(args.report, 'a') as f:
+            csv_writer = csv.writer(f)
+            csv_writer.writerow([generation_length, result_length, pipeline_opt_metrics.avg_cache_usage, pipeline_opt_metrics.max_cache_usage, rss_usage_gb])
```

Comment:
Should we pass  `generation_length` as a parameter to `run_and_write_metrics` or use `generation_config.max_new_tokens -1`?

Reply:
Done